### PR TITLE
React & Redux group project - Space Travelers' Hub

### PR DESCRIPTION
**Pull request by migo1**
### MILESTONES ACHIEVED FROM THE PROJECT

- Create React App.
- Install React Redux Redux Logger and React Router and other dependencies
- Create routes and view components: rockets, missions, my profile. Use <NavLink /> for the page navigation links and style active class to indicate which section/page the user is currently on (underline active navigation link).
- Create directories for all Redux state slice files: rockets, missions
- Upon the first render fetch data from the SpaceX API endpoints:
- Rockets: https://api.spacexdata.com/v4/rockets
- Missions: https://api.spacexdata.com/v3/missions
- Use useSelector() Redux Hook to select the state slices and render lists of rockets and missions in corresponding routes.
-   Write actions and reducers for booking/canceling rockets and joining/leaving missions.
- Render a list of all joined missions.
- Render a list of all reserved rockets.

### CODE REVIEW

File: src/redux/missions/missionsSlice.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  loading: true,
+  error: false,
+};
+
+export const getMissions = createAsyncThunk(
+  'spacex/getMissions',
+  async (thunkAPI) => {
+    try {
+      const resp = await axios.get('https://api.spacexdata.com/v3/missions');
+      return resp.data;
```
Comment (TedLivist) [inline src/redux/missions/missionsSlice.js]: - [ ] You are storing all mission data keys from the API, but according to the requirement, you should only save a few data from the API. Kindly refer to this [requirement section](https://github.com/microverseinc/curriculum-react-redux/blob/main/group-project/project_space_travelers_hub.md#redux-fetch-data-and-update-redux-store) to save only the needed data.

  image

File: src/redux/rockets/rocketsSlice.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+};
+
+const rocketsUrl = 'https://api.spacexdata.com/v3/rockets';
+
+export const getRockets = createAsyncThunk(
+  'rockets/getRockets',
+  async (thunkAPI) => {
+    try {
+      const resp = await axios.get(rocketsUrl);
+      return resp.data;
```
Comment (TedLivist) [inline src/redux/rockets/rocketsSlice.js]: - [ ] You are storing all rockets data keys from the API, but according to the requirement, you should only save a few data from the API. Kindly refer to this [requirement section](https://github.com/microverseinc/curriculum-react-redux/blob/main/group-project/project_space_travelers_hub.md#redux-fetch-data-and-update-redux-store) to save only the needed data.

  image

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @migo1 and @olanikegloria,

Good job so far!
✔️ Implementation profile page using `filter` method
✔️ Use of `<NavLink>` for navbar
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by migo1** [comment]
@TedLivist thanks for the review updatin code

File: src/redux/missions/missionsSlice.js (hunk 1 above)
Comment (Abenezer-Tilahun) [reply src/redux/missions/missionsSlice.js]: Fixed.

File: src/redux/rockets/rocketsSlice.js (hunk 2 above)
Comment (Abenezer-Tilahun) [reply src/redux/rockets/rocketsSlice.js]: Fixed.

**Review by Abenezer-Tilahun — APPROVED** [review]
Hi @migo1 
 
Your project is complete  👍 💯   There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 👏🏻

✔️ No linter errors.
✔️ You used correct Git-Flow.
✔️ Professional README  
✔️ Good Commit Messages
✔️ Good job for Implemented Suggested changes.

## Optional suggestions 
 
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

**Comment by migo1** [comment]
@Abenezer-Tilahun thanks for the review and approval

**Pull request merged**